packages/models-library/tests/test__models_examples.py

```diff
@@ -0,0 +1,59 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+
+
+import json
+from contextlib import suppress
+from importlib import import_module
+from inspect import getmembers, isclass
+from pathlib import Path
+from typing import Any, Iterable, Optional, Set, Tuple, Type
+
+import models_library
+import pytest
+from models_library.utils.misc import extract_examples
+from pydantic import BaseModel, NonNegativeInt
+from pydantic.json import pydantic_encoder
+
+
+def iter_model_cls_examples(
+    exclude: Optional[Set] = None,
+) -> Iterable[Tuple[str, Type[BaseModel], NonNegativeInt, Any]]:
+    def _is_model_cls(cls) -> bool:
+        with suppress(TypeError):
+            # NOTE: issubclass( dict[models_library.services.ConstrainedStrValue, models_library.services.ServiceInput] ) raises TypeError
+            return cls is not BaseModel and isclass(cls) and issubclass(cls, BaseModel)
+        return False
+
+    exclude = exclude or set()
+
+    for filepath in Path(models_library.__file__).resolve().parent.glob("*.py"):
+        if not filepath.name.startswith("_"):
+            mod = import_module(f"models_library.{filepath.stem}")
+            for name, model_cls in getmembers(mod, _is_model_cls):
+                if name in exclude:
+                    continue
+                # NOTE: this is part of utils.misc and is tested here
+                examples = extract_examples(model_cls)
+                for index, example in enumerate(examples):
+                    yield (name, model_cls, index, example)
+
+
+@pytest.mark.parametrize(
+    "class_name, model_cls, example_index, test_example", iter_model_cls_examples()
+)
+def test_all_module_model_examples(
```

Comment:
Q: hmm this is a rewrite of test_utils_misc right? is there something specific to know about it? or is it mostly just naming?

Reply:
now the parametrization distinguishes the ``test_example`` as well. Before they were all inside.
This is a very neat way to test all examples in all models defined in ``models_library``. 

I will probably create a helper in a separate PR to do that in all packages.

Reply:
what about using hypothesis?

Reply:
As discussed, it will be incorporated on a last stage